deploy-manage/users-roles/cluster-or-deployment-auth/controlling-access-at-document-field-level.md

```diff
@@ -109,6 +115,42 @@ POST /_security/role/dept_role
   ]
 }
 ```
+:::: 
+
+::::{tab-item} {{serverless-short}}
+:sync: serverless
+To configure document-level security (DLS), you create a custom role where you define the documents that this role grants access to, using the [QueryDSL](/explore-analyze/query-filter/languages/querydsl.md) syntax:
+
+1. Go to the **Custom Roles** page using the navigation menu or the [global search field](/explore-analyze/find-and-organize/find-apps-and-objects.md).
+1. Select **Create role**.
+1. Give your custom role a meaningful name and description.
+1. In the **Index privileges** area, specify the data stream pattern and the privilege you want to grant. For example, enter `events-*` and `read`.
+1. Enable the **Grant read privileges to specific documents** toggle and add your query using the QueryDSL syntax.
+
+    * For example, to allow read access only to documents that belong to the click category within all the `events-*` data streams, enter the following query:
+        ```
+        {
+          "match" : { "category" : “click” }
+        }
+        ```
+
+        ![Configuring document-level security](/deploy-manage/images/serverless-custom-role-document-level-privileges-ex-1.png)
+        
+    * To allow read access only to the documents whose `department_id` equals 12, enter the following query:
+        ```
+        {
+          "term" : { "department_id" : 12 }
+        }
+        ```
+
+        ![Configuring document-level security another example](/deploy-manage/images/serverless-custom-role-document-level-privileges-ex-2.png)
+        
+1. Optional: To grant this role access to {{kib}} spaces for feature access and visibility, click **Assign to this space**. Specify the level of access required and click **Assign role**.
+1. Select **Create role** to save your custom role.
+::::
+
+:::::
+
 
 ### Templating a role query [templating-role-query]
```

Comment:
does this work for serverless? wonder if these variables are available. if it does work, it would be nice to just tab the code examples (one with the query, one with just the query body)

Reply:
[As far as I can tell](https://www.elastic.co/docs/api/doc/elasticsearch-serverless/operation/operation-security-put-role#operation-security-put-role-body-application-json-indices), you can use these variables:
<img width="976" height="442" alt="image" src="https://github.com/user-attachments/assets/b983d5e3-fbf9-4b36-a910-eeb8359a1151" />

I've tried to add tabs to the code examples, but am stuck as I don't know how to verify them. Here's the preview for the [Templating a role query](https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/3950/deploy-manage/users-roles/cluster-or-deployment-auth/controlling-access-at-document-field-level#templating-role-query) section.

Is this what you had in mind, @shainaraskas ?